File: frontend/src/api/landing.ts
import axios from '@/http/axios';

export interface NewsletterSubscriptionRequest {
  email: string;
  name?: string;
  source?: string;
  marketing_consent?: boolean;
}

export interface ContactFormRequest {
  name: string;
  email: string;
  subject: string;
  message: string;
  company?: string;
  phone?: string;
}

export interface LandingResponse {
  success: boolean;
  message: string;
  timestamp: string;
  data?: Record<string, any>;
}

export interface LandingStats {
  total_subscribers: number;
  total_contacts: number;
  conversion_rate: number;
  last_updated: string;
}

export interface Feature {
  title: string;
  description: string;
  icon: string;
  category?: string;
}

export interface Testimonial {
  name: string;
  company: string;
  text: string;
  rating: number;
  avatar_url?: string;
}

export interface LandingFeatures {
  features: Feature[];
}

export interface LandingTestimonials {
  testimonials: Testimonial[];
}

export const subscribeToNewsletter = async (request: NewsletterSubscriptionRequest): Promise<LandingResponse> => {
  const response = await axios.post('/landing/newsletter', request);
  return response.data;
};

export const submitContactForm = async (request: ContactFormRequest): Promise<LandingResponse> => {
  const response = await axios.post('/landing/contact', request);
  return response.data;
};

export const getLandingStats = async (): Promise<LandingStats> => {
  const response = await axios.get('/landing/stats');
  return response.data;
};

export const getLandingFeatures = async (): Promise<LandingFeatures> => {
  const response = await axios.get('/landing/features');
  return response.data;
};

export const getLandingTestimonials = async (): Promise<LandingTestimonials> => {
  const response = await axios.get('/landing/testimonials');
  return response.data;
};

export const getLandingPlans = async (): Promise<unknown[]> => {
  const response = await axios.get('/landing/plans');
  return response.data;
}; 

File: backend/schemas/landing.py
"""
Landing Page Schemas
Pydantic models for landing page API endpoints
"""

from datetime import datetime

from pydantic import BaseModel, EmailStr


class NewsletterSubscription(BaseModel):
    """Newsletter subscription model"""

    email: EmailStr
    name: str | None = None
    source: str | None = "landing_page"
    marketing_consent: bool = True
    subscribed_at: datetime | None = None


class ContactForm(BaseModel):
    """Contact form model"""

    name: str
    email: EmailStr
    subject: str
    message: str
    company: str | None = None
    phone: str | None = None
    submitted_at: datetime | None = None


class LandingResponse(BaseModel):
    """Standard landing page response"""

    success: bool
    message: str
    timestamp: datetime
    data: dict | None = None


class LandingStats(BaseModel):
    """Landing page statistics"""

    total_subscribers: int
    total_contacts: int
    conversion_rate: float
    last_updated: datetime


class Feature(BaseModel):
    """Platform feature for landing page"""

    title: str
    description: str
    icon: str
    category: str | None = None


class Testimonial(BaseModel):
    """Customer testimonial"""

    name: str
    company: str
    text: str
    rating: int
    avatar_url: str | None = None


class LandingFeatures(BaseModel):
    """Landing page features response"""

    features: list[Feature]


class LandingTestimonials(BaseModel):
    """Landing page testimonials response"""

    testimonials: list[Testimonial]


# Admin schemas for content management
class LandingSectionCreate(BaseModel):
    """Create/update landing section"""
    section_type: str
    content: dict
    is_active: bool = True
    order_index: int = 0


class LandingSectionResponse(BaseModel):
    """Landing section response"""
    id: str
    section_type: str
    content: dict
    is_active: bool
    order_index: int
    updated_at: datetime | None = None


class LandingFeatureCreate(BaseModel):
    """Create/update landing feature"""
    title: str
    description: str
    icon: str
    tech_stack: str | None = None
    gradient: str | None = None
    animation: str | None = None
    is_active: bool = True
    order_index: int = 0


class LandingFeatureResponse(BaseModel):
    """Landing feature response"""
    id: str
    title: str
    description: str
    icon: str
    tech_stack: str | None = None
    gradient: str | None = None
    animation: str | None = None
    is_active: bool
    order_index: int


class LandingPricingCreate(BaseModel):
    """Create/update pricing plan"""
    name: str
    price: str
    period: str
    description: str
    features: list[str]
    icon: str | None = None
    is_popular: bool = False
    badge_text: str | None = None
    cta_text: str = "Get Started"
    is_active: bool = True
    order_index: int = 0


class LandingPricingResponse(BaseModel):
    """Pricing plan response"""
    id: str
    name: str
    price: str
    period: str
    description: str
    features: list[str]
    icon: str | None = None
    is_popular: bool
    badge_text: str | None = None
    cta_text: str
    is_active: bool
    order_index: int


class LandingContentResponse(BaseModel):
    """Complete landing page content"""
    sections: dict[str, dict]
    features: list[LandingFeatureResponse]
    pricing: list[LandingPricingResponse]


class LandingLeadResponse(BaseModel):
    """Landing lead response"""
    id: str
    email: str
    name: str | None = None
    company: str | None = None
    phone: str | None = None
    source: str
    created_at: datetime
    converted_to_user: bool
